.github/workflows/release.yml

```diff
@@ -0,0 +1,36 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - master
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    environment: NPM
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+
+      - name: Install Dependencies
+        run: yarn --frozen-lockfile
+
+      - name: Create Release Pull Request or Publish to NPM
+        uses: changesets/action@v1
+        with:
+          publish: yarn release
+          commit: "chore(release): publish packages"
+          title: "Publish packages"
+          createGithubRelease: true
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```

Comment:
This NPM token is for the @near-js NPM organization so will not work for publishing near-api-js. This token will need to be used as we start to pull out/create new packages, but if we want CI for near-api-js right now, we'll need to use a token capable of publishing that package.